views/list_referencing_type_counts.aql

```diff
@@ -0,0 +1,21 @@
+// Fetch the counts of a ws_objects in the RE that is linked to a wsprov_object
+// Args:
+//   ws_ids - array of private workspace ids the user has access to (autofilled by RE-API)
+//   key - key of the object to start from
+//   is_private - if present, limit to objects in workspaces that a user has access to
+//   is_public - if present, limit to objects in public workspaces
+//   *** if both is_private and is_public are true this will be treated as an OR ***
+//   owners - if present, limit to objects with owner in list
+//   simplify_type - if true, strip out the module and version before collecting
+
+WITH wsprov_object
+FOR v IN 1..10
+    INBOUND CONCAT('wsprov_object/', @key) wsprov_links
+    OPTIONS {'uniqueVertices': 'global', 'bfs':true}
+    // If both is_private and is_public, return if either is true
+    FILTER (@is_private && @is_public) ? (v.is_public || v.workspace_id IN @ws_ids) :
```

Comment:
this could be a reason for jinja templating at some point in the future. we could remove the ternary through templating, which might speed it up

Reply:
The ternary is pretty quick, I think the main thing that would be improved is support for optional keys

Reply:
doing `... : (@is_private && v.workspace_id IN @ws_ids) || (@is_public && v.is_public)` might be more readable. Also, setting `is_public` to false can still show objects where `is_public` is true, I think

Reply:
Here's how it might work with templating:
```
{% if is_public and is_private %} FILTER v.is_public || v.workspace_id in @ws_ids {% endif %}
{% if is_public and not is_private %} FILTER v.is_public {% endif %}
{% if not is_public and is_private %} FILTER v.workspace_id in @ws_ids && !v.is_public {% endif %}
```

Reply:
Yeah, that's how I had designed it but I could change it to exclude private or public results if the value is 0 being as it's not really optional

Reply:
I'm not sure I followyou here.. `is_public` can be required but still set to `false`, right?